lms/static/js/discussions_management/views/discussions.js

```diff
@@ -16,11 +28,148 @@
+                    self.$('.division-scheme-container').prepend(this.notification.$el);
+                    this.notification.render();
+                },
+
+                removeNotification: function() {
+                    if (this.notification) {
+                        this.notification.remove();
+                    }
+                },
+
+                getSelectedScheme: function() {
+                    return this.$('input[name="division-scheme"]:checked').val();
+                },
+
+                getTopicNav: function() {
+                    return this.$('.topic-division-nav');
+                },
+
+                divisionSchemeChanged: function() {
+                    var selectedScheme = this.getSelectedScheme(),
+                        topicNav = this.getTopicNav(),
+                        messageSpan = this.$('.division-scheme-message'),
+                        fieldData = {
+                            division_scheme: selectedScheme
+                        };
+
+                    this.hideTopicNav(selectedScheme, topicNav);
+                    this.saveDivisionScheme(topicNav, fieldData);
+                    this.showSelectMessage(selectedScheme, messageSpan);
+                },
+
+                saveDivisionScheme: function($element, fieldData) {
+                    var self = this,
+                        discussionSettingsModel = this.discussionSettings,
+                        saveOperation = $.Deferred(),
+                        showErrorMessage;
+
+                    this.removeNotification();
+                    showErrorMessage = function(message) {
+                        self.showMessage(message, 'error');
+                    };
+
+                    discussionSettingsModel.save(
+                        fieldData, {patch: true, wait: true}
+                    ).done(function() {
+                        saveOperation.resolve();
+                        self.showMessage(
+                            gettext('Your changes have been saved.')
+                        );
+                    }).fail(function(result) {
+                        var errorMessage = null,
+                            jsonResponse;
+                        try {
+                            jsonResponse = JSON.parse(result.responseText);
+                            errorMessage = jsonResponse.error;
+                        } catch (e) {
+                            // Ignore the exception and show the default error message instead.
+                        }
+                        if (!errorMessage) {
+                            errorMessage = gettext('We\'ve encountered an error. ' +
+                                'Refresh your browser and then try again.');
+                        }
+                        showErrorMessage(errorMessage);
+                        saveOperation.reject();
+                    });
+                },
+
+                hideTopicNav: function(selectedScheme, topicNav) {
```

Comment:
The method name implies that the topic nav is always hidden. Perhaps updateTopicNavigationVisibility or showOrHideTopicNav or setTopicNavVisibility?